### PR TITLE
Add missing jshint dependency

### LINKED ISSUE #65: Trying to run tests gives an error
When running `npm run test` I get this:

```
> umzug@1.9.1 test /home/bjn/src/umzug
> gulp

module.js:341
    throw err;
    ^

Error: Cannot find module 'jshint/src/cli'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (/home/bjn/src/umzug/node_modules/gulp-jshint/src/extract.js:1:79)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17
```

### PR DESCRIPTION
**Pull request by tremby**
Closes #65

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "main": "index.js",
   "dependencies": {
     "bluebird": "^3.0.1",
+    "jshint": "^2.9.1",
```
Comment (jukkah) [inline package.json]: This should be in `devDependencies` and in addition not required at all. gulp-jshint@^2.0.0 has jshint@2.x in `peerDependencies`. It should be enough.

Comment (tremby) [inline package.json]: Peer dependencies don't work that way, at least not in npm 3.7.3.

```
[bjn@spiral:~/src/umzug]$ rm -rf node_modules
[bjn@spiral:~/src/umzug]$ npm install
...
umzug@1.9.1 /home/bjn/src/umzug
├── bluebird@3.3.4 
├── coffee-script@1.10.0
...
│ │ ├── os-tmpdir@1.0.1 
│ │ └── rimraf@2.2.8 
│ └── through@2.3.8 
├── UNMET PEER DEPENDENCY jshint@2.x
├─┬ jshint-stylish@2.1.0 
│ ├── log-symbols@1.0.2 
│ ├── string-length@1.0.1 
│ └── text-table@0.2.0 
...
  └── yargs-parser@2.1.2 

npm WARN gulp-jshint@2.0.0 requires a peer of jshint@2.x but none was installed.
```

But regarding devDependencies -- true. I'll fix that.

Comment (tremby) [inline package.json]: See https://github.com/npm/npm/issues/6565

**Comment by jukkah** [comment]
LGTM

**Pull request merged**